### PR TITLE
Optional keyword argument ignore_zeros for mean_absolute_percentage_e…

### PR DESCRIPTION
**Pull request by vibhamasti**
Option to ignore zero `y_true` values while calculating `mean_absolute_percentage_error` in `keras.metrics.mean_absolute_percentage_error` to prevent unnecessarily large values

### CODE REVIEW

**Comment by vibhamasti** [comment]
I request a repo manager to rescan the PR as I just signed the CLA.

**Review by Fii02kong — APPROVED** [review]

File: keras/losses.py [hunk 1]
```diff
@@ -1468,11 +1469,14 @@ def _ragged_tensor_mae(y_true, y_pred):
               'keras.losses.mean_absolute_percentage_error',
               'keras.losses.mape', 'keras.losses.MAPE')
 @tf.__internal__.dispatch.add_dispatch_support
-def mean_absolute_percentage_error(y_true, y_pred):
+def mean_absolute_percentage_error(y_true, y_pred, ignore_zeros):
```
Comment (qlzh727) [inline keras/losses.py:1472]: Please make ignore_zeros a keyword arg so that it will have a default value (which should be False), otherwise it will break all the existing caller to this API.

File: keras/losses.py [hunk 2]
```diff
@@ -1468,11 +1469,14 @@ def _ragged_tensor_mae(y_true, y_pred):
...
   """Computes the mean absolute percentage error between `y_true` and `y_pred`.
 
   `loss = 100 * mean(abs((y_true - y_pred) / y_true), axis=-1)`
 
+  If `ignore_zeros` is set to True, the calculation ignores values with 0. Otherwise,
```
Comment (qlzh727) [inline keras/losses.py:1477]: Please be specific about what 0 is ignored, is it from the prediction or the label?

File: keras/losses.py [hunk 3]
```diff
@@ -1468,11 +1469,14 @@ def _ragged_tensor_mae(y_true, y_pred):
...
+  a small epsilon value is used to avoid division with 0.
+
```
Comment (qlzh727) [inline keras/losses.py:1479]: Please provide a sample use case for ignore_zeros = True (and about why user should use it).

Comment (qlzh727) [reply keras/losses.py:1479]: Please also update the docstring for the ignore_zero below.

File: keras/losses.py [hunk 4]
```diff
@@ -1496,6 +1500,10 @@ def mean_absolute_percentage_error(y_true, y_pred):
   diff = tf.abs(
       (y_true - y_pred) / backend.maximum(tf.abs(y_true),
                                           backend.epsilon()))
+  if ignore_zeros is True:
```
Comment (qlzh727) [inline keras/losses.py:1503]: Just use "if ignore_zeros" (remove the is True check). The implicit check is preferred and more readable in this case.

File: keras/losses.py [hunk 5]
```diff
@@ -1496,6 +1500,10 @@ def mean_absolute_percentage_error(y_true, y_pred):
...
+    mask = y_true != 0
```
Comment (qlzh727) [inline keras/losses.py:1504]: You might want to add some test for different dtypes (floats), I would expect `y_true != 0` might raise some error when dtype doesn't match.

File: keras/losses.py [hunk 6]
```diff
@@ -1496,6 +1500,10 @@ def mean_absolute_percentage_error(y_true, y_pred):
...
+    diff = diff[mask]
```
Comment (qlzh727) [inline keras/losses.py:1505]: Is this intended? It will fill some 0s to diff, and those zeros are still taking into account when it does the mean. From the docstring above, I thought you are going to "ignore" the zeros.

**Review by qlzh727 — CHANGES_REQUESTED** [review]
It was unclear to me what's the use case for this new param.

Could you also add some unit test for this new change?

**Comment by fchollet** [comment]
I believe you can implement this functionality in a cleaner fashion by using `sample_weight` and setting a weight of 0 for any 0 values. Have you tried it? If possible we should seek not to add a new argument here.

**Comment by gbaned** [comment]
@vibhamasti Can you please check @qlzh727's comments and keep us posted ? Thanks!
